MarvelApiClient/src/main/java/com/karumi/MarvelApiClient/HashGenerator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *   Copyright (C) 2015 Karumi.
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.karumi.marvelapiclient;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+class HashGenerator {
```

Comment:
This class name is not as specific as should. This hash generator is just valid for this MarvelApiClient, my suggestion is to rename this class to ``MarvelAuthHashGenerator``.